Convert testing values to basis points

brianmcmichael opened the pull request:
We were testing against percentages multiplied by 1000, I think it makes more sense to test against basis points, so I updated the tests.

brianmcmichael committed:
Convert testing values to basis points

kmbarry1 commented on src/Kovan-DssSpell.t.sol:
```diff
@@ -454,12 +452,12 @@ contract DssSpellTest is DSTest, DSMath {
         // 1000000073014496989316680335
         assertTrue(duty >= RAY && duty < 1000000073014496989316680335);  // gt 0 and lt 1000%
-        assertTrue(diffCalc(expectedRate(values.collaterals[ilk].pct), yearlyYield(rates.rates(values.collaterals[ilk].pct / 10))) <= TOLERANCE);
+        assertTrue(diffCalc(expectedRate(values.collaterals[ilk].pct * 10), yearlyYield(rates.rates(values.collaterals[ilk].pct))) <= TOLERANCE);
```
same comment as other PR...basis points all the way down would be nice, but again no need to block on it if you disagree

brianmcmichael replied:
I was just making the smallest incision here. I can look into taking it the rest of the way.

kmbarry1 replied:
Sorry, looks like I didn't communicate clearly enough--I just meant that it was very inelegent to be multiplying `pct` by 10 here, so it would be nice if `expectedRate` could just expect a basis point argument, I didn't mean to use basis points to express all values.

brianmcmichael replied:
That's fine. Pushed a fix for this. The other changes are big improvements IMO since they are easier to inspect and reason about now and provide some addition protection against transcription errors.

kmbarry1 reviewed (approved):
LGTM

brianmcmichael committed:
convert chop and mat to bp

brianmcmichael committed:
Convert test beg to bp

brianmcmichael committed:
Convert dunk to whole units

brianmcmichael committed:
Convert line to Dai units

brianmcmichael committed:
Convert dust to dai units

brianmcmichael committed:
Dsr to BP

brianmcmichael committed:
Line to whole units

brianmcmichael committed:
dump and sump

brianmcmichael committed:
Normalize bump, hump, box and add count check

brianmcmichael commented:
Updated to use basis points for all values, updated all values to use Dai units.

This should eliminate many potential copy/paste errors when values are getting updated weekly and values are easier on human auditing.

brianmcmichael committed:
Update expectedRate funtion

gbalabasquer commented on src/Kovan-DssSpell.t.sol:
```diff
@@ -169,197 +167,197 @@ contract DssSpellTest is DSTest, DSMath {
-            vow_bump: 10 * RAD,
-            vow_hump: 500 * RAD,
-            cat_box: 10 * THOUSAND * RAD
+            dsr_rate:     0,               // In basis points
```
I'd just rename it back to `pot_dsr` to keep the same naming convention of the rest of General system variables.
